### PR TITLE
Feat/refactor store

### PR DESCRIPTION
**Pull request by developer0623**

### CODE REVIEW

File: src/components/ColumnsDisplay/index.tsx [hunk 1]
```diff
@@ -169,22 +177,13 @@ const styles = {
...
-
 type Props = {
-  selectedColumns: any;
+  selectedColumns: any[];
   selectedEntity: string;
   attributes: any;
   classes: any;
   submitValues: () => void;
-  setColumns: (columns: object[]) => void;
+  setColumns: (entity: string, columns: object[]) => void;
```
Comment (anonymoussprocket) [inline src/components/ColumnsDisplay/index.tsx]: If we're all in on Typescript, should all `object` become `any`?

File: src/components/FilterPanel/index.tsx [hunk 2]
```diff
@@ -96,140 +94,94 @@ const attrTabValue = {
 interface Filter {
   name: string;
   operator: string;
+  operatorType: string;
+  isCard?: boolean;
```
Comment (anonymoussprocket) [inline src/components/FilterPanel/index.tsx]: What is the meaning of "card" here?

Comment (anonymoussprocket) [reply src/components/FilterPanel/index.tsx]: please rename to `isLowCardinality`

File: src/components/FilterSelect/index.tsx [hunk 3]
```diff
@@ -181,6 +173,7 @@ class FilterSelect extends React.Component<Props, States> {
           aria-haspopup="true"
           isactive={value}
           onClick={this.handleClick}
+          iscapital={placeholder !== 'Select Operator'? 1 : 0}
```
Comment (anonymoussprocket) [inline src/components/FilterSelect/index.tsx]: What's `iscapital` here? Seems odd to compare to a constant. That constant also should be named, not in-line.

Comment (anonymoussprocket) [reply src/components/FilterSelect/index.tsx]: If this is to format the operator values, that should be done elsewhere, like a function that returns operators for a specific datatype.

File: src/containers/App/index.tsx [hunk 4]
```diff
@@ -114,38 +108,27 @@ const tabsArray = [
...
 }
 
 class Arronax extends React.Component<Props, States> {
   constructor(props: Props) {
     super(props);
     this.state = {
-      isFilterCollapse: false,
-      filterInputState: { blocks: [], operations: [], accounts: [] },
-      selectedDisplayColumns: [],
+      isFilterCollapse: false
```
Comment (anonymoussprocket) [inline src/containers/App/index.tsx]: rename to `isFilterCollapsed`

File: src/reducers/app/reducers.ts [hunk 5]
```diff
@@ -30,7 +28,7 @@ export interface AppState {
   filters: ConseilQuery;
   network: string;
   blocks: TezosBlock[];
-  availableValues: Array<string>;
+  availableValues: object;
```
Comment (anonymoussprocket) [inline src/reducers/app/reducers.ts]: `any`?

File: src/reducers/app/reducers.ts [hunk 6]
```diff
@@ -93,13 +90,16 @@ const initialState: AppState = {
     operations: [],
     accounts: [],
   },
-  availableValues: [],
+  availableValues: {
```
Comment (anonymoussprocket) [inline src/reducers/app/reducers.ts]: ideally this should come from the metadata service

File: src/reducers/app/reducers.ts [hunk 7]
```diff
@@ -121,10 +121,9 @@ const appReducer = (state = initialState, action) => {
     case SET_ITEMS:
       return { ...state, [action.entity]: action.items };
     case SET_COLUMNS: {
-      const columns = state.columns;
-      const newColumns = { ...columns };
-      newColumns[action.entity] = action.items;
-      return { ...state, columns: newColumns };
+      const columns = {...state.columns};
```
Comment (anonymoussprocket) [inline src/reducers/app/reducers.ts]: Why not `const columns = state.columns;`?

File: src/reducers/app/reducers.ts [hunk 8]
```diff
@@ -135,75 +134,49 @@ const appReducer = (state = initialState, action) => {
...
-      const attributes = state.attributes;
+      const attributes = { ...state.attributes};
       attributes[action.entity] = action.attributes;
       return { ...state, attributes };
     }
     case ADD_FILTER: {
-      const selectedFilters = state.selectedFilters;
+      const selectedFilters = {...state.selectedFilters};
       let filters = selectedFilters[action.entity];
       const emptyFilter = {
```
Comment (anonymoussprocket) [inline src/reducers/app/reducers.ts]: Create a `Filter` type

File: src/components/FilterPanel/index.tsx [hunk 9]
```diff
@@ -96,140 +94,94 @@ const attrTabValue = {
...
 }
 
 type Props = {
-  selectedValues: object[];
-  availableValues: object[];
+  availableValues: object;
   selectedEntity: string;
   attributes: any[];
   filters: Array<Filter>;
   operators: any;
```
Comment (anonymoussprocket) [inline src/components/FilterPanel/index.tsx]: Reuse the ConseilOperator enum from ConseilJS

File: src/components/FilterPanel/index.tsx [hunk 10]
```diff
@@ -96,140 +94,94 @@ const attrTabValue = {
...
     } = this.props;
 
-    const cards = attributes.map(attr => {
-      if (attr.cardinality < 15 && attr.cardinality !== null) {
-        return attr.name;
-      }
-    });
-    if (cards.includes(val)) {
-      fetchValues(val);
+    const isCard = attr.cardinality < 15 && attr.cardinality !== null;
```
Comment (anonymoussprocket) [inline src/components/FilterPanel/index.tsx]: `15` should be a named constant

File: src/components/FilterPanel/index.tsx [hunk 11]
```diff
@@ -96,140 +94,94 @@ const attrTabValue = {
...
-    selectedFilter.name = val;
+    let operatorType = 'dateTime';
+    if (attr.dataType === 'Int' || attr.dataType === 'Decimal') {
+      operatorType = 'numeric';
+    } else if (attr.dataType === 'String') {
+      operatorType = 'string';
+    } else if (attr.dataType === 'Boolean') {
+      operatorType = 'boolean';
+    }
+    const selectedFilter = {
```
Comment (anonymoussprocket) [inline src/components/FilterPanel/index.tsx]: Use a type here

File: src/components/FilterSelect/index.tsx [hunk 12]
```diff
@@ -16,7 +16,7 @@ const ButtonShell = styled(Button)`
     font-size: 18px;
     cursor: pointer;
     color: ${({ isactive }) => (isactive ? '#4A4A4A' : '#9b9b9b')};
```
Comment (anonymoussprocket) [inline src/components/FilterSelect/index.tsx]: Use CSS references

File: src/components/ValueInput/index.tsx [hunk 13]
```diff
@@ -32,100 +32,63 @@ const TextInput = styled(TextField)`
...
-    const findStateValue = filterInputState[selectedEntity].find(
-      value => Object.keys(value).toString() === filter.name
-    );
-    const currentStateValue = findStateValue
-      ? Object.values(findStateValue).toString()
-      : '';
     // Render specific input type based on operators
-    if (filter.operator === 'BETWEEN' || filter.operator === 'IN') {
-      const splitValues = currentStateValue.split('-');
+    if (operator === 'BETWEEN' || operator === 'IN') {
```
Comment (anonymoussprocket) [inline src/components/ValueInput/index.tsx]: Use the same enum as mentioned above for this.

File: src/reducers/app/thunks.ts [hunk 14]
```diff
@@ -35,6 +35,20 @@ const SYNC_TIME = 10;
 const configs = getConfigs();
 const { getAttributes, getAttributeValues } = ConseilMetadataClient;
 
+const ConseilOperators = {
```
Comment (anonymoussprocket) [inline src/reducers/app/thunks.ts]: Import `ConseilOperator` from ConseilJS `QueryTypes`

File: src/components/ColumnsDisplay/index.tsx (hunk 1 above)
Comment (developer0623) [reply src/components/ColumnsDisplay/index.tsx]: @anonymoussprocket we need to use the correct type.

File: src/components/FilterSelect/index.tsx (hunk 3 above)
Comment (developer0623) [reply src/components/FilterSelect/index.tsx]: capital

Comment (developer0623) [reply src/components/FilterSelect/index.tsx]: We used the same component for the two selectors.
When it is for the operator, it should display a small character.

File: src/reducers/app/reducers.ts (hunk 6 above)
Comment (developer0623) [reply src/reducers/app/reducers.ts]: Right, we are getting the values using `getAttributeValues` function of the  `ConseilMetadataClient`, now.

File: src/reducers/app/reducers.ts (hunk 7 above)
Comment (developer0623) [reply src/reducers/app/reducers.ts]: `const columns = state.columns;`
The reference of `columns` and `state.columns` is the same.
That is, no new variables were actually created.
Destructuring assignment creates a new variable.

File: src/components/ColumnsDisplay/index.tsx (hunk 1 above)
Comment (anonymoussprocket) [reply src/components/ColumnsDisplay/index.tsx]: This is for later:

https://github.com/Cryptonomic/Arronax/issues/158

File: src/components/FilterSelect/index.tsx (hunk 3 above)
Comment (anonymoussprocket) [reply src/components/FilterSelect/index.tsx]: `ConseilOperator` `enum` in ConseilJS has these in lower case, but you'd still need to format "startWith" into "starts with". Maybe implement a utility function for this that takes a data type as parameter and returns the display values or display tuples like `[{'startsWIth': 'starts with'}, 'in', 'in']`, etc.

File: src/reducers/app/reducers.ts (hunk 6 above)
Comment (anonymoussprocket) [reply src/reducers/app/reducers.ts]: I meant use `getEntities()`

**Pull request merged**